devito/types/dimension.py

```diff
@@ -536,6 +535,81 @@ def _arg_check(self, *args, **kwargs):
 # The Dimensions below are exposed in the user API. They can only be created by
 # the user
 
+class SubDimensionThickness(DataSymbol):
```

Comment:
this should be called just "Thickness", it's a first-class citizen now

Reply:
Maybe I should swap the `namedtuple` "Thickness" and `SubDomainThickness` class names?